core/src/apps/cardano/layout.py

```diff
@@ -195,18 +196,49 @@ async def show_address(
     Custom show_address function is needed because cardano addresses don't
     fit on a single screen.
     """
+    #path = [1852 | HARDENED, 1815 | HARDENED, 100000 | HARDENED, 0, 100000]
+
+    def addr_str_len(i: int)-> int:
+        if i & HARDENED:
+            return len(str(i^HARDENED)) + 2
+        return len(str(i)) + 1
+
     path_str = address_n_to_str(path)
-    t1 = Text(path_str, ui.ICON_RECEIVE, ui.GREEN)
+    t1 = Text("%s address" % ADDRESS_TYPE_NAMES[address_type], ui.ICON_RECEIVE, ui.GREEN)
+
+    used_lines = 1
+
     if network is not None:
         t1.normal("%s network" % protocol_magics.to_ui_string(network))
-    t1.normal("%s address" % ADDRESS_TYPE_NAMES[address_type])
+        used_lines += 1
+
+    if len(path_str) > 22:
```

Comment:
why not just `t1.normal(path_str)`? This is not the only place where we are displaying the path on the main screen and I see no similar line-wrapping logic.

wrapping the path at `/` is nice but IMHO not worth the hassle. I tried rendering a long path and it looks like this

![Untitled](https://user-images.githubusercontent.com/4980147/93631920-b73c4580-f9ec-11ea-8d36-acb5365e4876.png)

which looks good to me - it serves the purpose of displaying the full path to the user. The line may break just at the `'` character but as far as I tested if the `'` character is the only one to overflow, the native trezor line-wrapping logic takes care of that and puts the hyphen one character earlier as can be seen here:

![Untitled](https://user-images.githubusercontent.com/4980147/93633849-e99b7200-f9ef-11ea-890d-56b742a53c83.png)

Or did you find another corner case?

Reply:
also, if I were to implement DYI line wrapping, I would rather take an approach similar to the one outlined here: https://stackoverflow.com/questions/32122022/split-a-string-into-pieces-of-max-length-x-split-only-at-spaces - basically just an accumulator variable which would be taking the path items one by one and if the accumulator were to overflow the line, it would print the accumulator content, starting over with the rest of items, but I would rather not do that at all unless really needed. The logic here is imho too hard to understand and have a grasp of the different cornercases it's trying to tackle

Reply:
main reason i kept this code is, that you dont know how many lines does the address path print, which means you dont know how many lines of the actual address to print